banzai/qc/header_checker.py

```diff
@@ -0,0 +1,155 @@
+    """
+       This class contains the needed function/definitions to performs the header check.
+    """
+
+    def __init__(self, pipeline_context):
+        super(HeaderSanity, self).__init__(pipeline_context)
+
+        self.header_expected_format = {'RA': str, 'DEC': str, 'CAT-RA': str, 'CAT-DEC': str,
+                                       'OFST-RA': str, 'OFST-DEC': str, 'TPT-RA': str,
+                                       'TPT-DEC': str, 'PM-RA': str, 'PM-DEC': str,
+                                       'CRVAL1': float, 'CRVAL2': float, 'CRPIX1': int,
+                                       'CRPIX2': int, 'EXPTIME': float}
+
+    @property
+    def group_by_keywords(self):
+        return None
+
+    def do_stage(self, images):
+        """ Performs several checks on the header, see the relative functions.
+        :param list images: a list of image object.
+        :returns: the list of image object after check
+        :rtype: list
+        """
+        for image in images:
+
+
+            for keyword in self.header_expected_format.keys():
+                self.check_header_keyword_present(keyword, image)
+                self.check_header_format(keyword, image)
+                self.check_header_na(keyword, image)
+
+            self.check_ra_range(image)
+            self.check_dec_range(image)
+            self.check_exptime_value(image)
+
+        return images
+
+    def check_header_keyword_present(self, keyword, image):
+        """ Return a warning if the keyword is not in the image header.
+        :param str keyword: the stry keyword you look inside the header
+        :param object image: an image object
+        """
+        header = image.header
+
+        logging_tags = logs.image_config_to_tags(image, self.group_by_keywords)
+
+        if keyword not in header:
+            sentence = 'WARNING : The header key ' + keyword + ' is not in image header!'
+
+            self.logger.error(sentence, extra=logging_tags)
+
+            return 'WARNING'
+        return
+
+    def check_header_format(self, keyword, image):
+        """ Return a warning if the keyword is not the expected type
+            in the image header.
+        :param str keyword: the stry keyword you look inside the header
+        :param object image: an image object
+        """
+        header_value = image.header[keyword]
+
+        logging_tags = logs.image_config_to_tags(image, self.group_by_keywords)
+
+        if not isinstance(header_value, self.header_expected_format[keyword]):
+            sentence = ('The header key ' + keyword + ' got an unexpected format : ' + type(
+                header_value).__name__ + ' in place of ' + self.header_expected_format[
+                    keyword].__name__)
+
+            self.logger.error(sentence, extra=logging_tags)
+
+        return
+
+    def check_header_na(self, keyword, image):
+        """ Return a warning if the keyword is 'N/A' instead of the
+            expected type in the image header.
```

Comment:
Does these BIAS and DARK have specificity inside OBSTYPE for example? If yes, I can put an exception.